doc/source/data/batch_inference.rst

```diff
@@ -0,0 +1,752 @@
+Below you find examples for PyTorch, TensorFlow, and HuggingFace.
+
+.. tabs::
+
+    .. group-tab:: HuggingFace
+
+        .. callout::
+
+            .. literalinclude:: ./doc_code/hf_quick_start.py
+                :language: python
+                :start-after: __hf_quickstart_model_start__
+                :end-before: __hf_quickstart_model_end__
+
+            .. annotations::
+                <1> Use the constructor (``__init__``) to initialize your model.
+
+                <2> The ``__call__`` method runs inference on a batch of data.
+
+    .. group-tab:: PyTorch
+
+        .. callout::
+
+            .. literalinclude:: ./doc_code/pytorch_quick_start.py
+                :language: python
+                :start-after: __pt_quickstart_model_start__
+                :end-before: __pt_quickstart_model_end__
+
+            .. annotations::
+                <1> Use the constructor (``__init__``) to initialize your model.
+
+                <2> The ``__call__`` method runs inference on a batch of data.
+
+
+    .. group-tab:: TensorFlow
+
+        .. callout::
+
+            .. literalinclude:: ./doc_code/tf_quick_start.py
+                :language: python
+                :start-after: __tf_quickstart_model_start__
+                :end-before: __tf_quickstart_model_end__
+
+            .. annotations::
+                <1> Use the constructor (``__init__``) to initialize your model.
+
+                <2> The ``__call__`` method runs inference on a batch of data.
+
+
+3. Getting Predictions with Ray Data
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once you have your Ray Datastream ``ds`` and your predictor class, you can use
+``ds.map_batches(...)`` to get predictions.
+Mapping batches this way is the recommended way to run inference with Ray, but there are also other options.
+``map_batches`` takes your predictor class as an argument and allows you to specify
+``compute`` resources by defining a so-called ``ActorPoolStrategy``.
+In the example below, we use two CPUs to run inference in parallel and then print the results.
+We cover resource allocation in more detail in :ref:`the configuration section of this guide<batch_inference_config>`.
+
+.. tabs::
+
+    .. group-tab:: HuggingFace
+
+        .. literalinclude:: ./doc_code/hf_quick_start.py
+            :language: python
+            :start-after: __hf_quickstart_prediction_start__
+            :end-before: __hf_quickstart_prediction_end__
+
+    .. group-tab:: PyTorch
+
+        .. literalinclude:: ./doc_code/pytorch_quick_start.py
+            :language: python
+            :start-after: __pt_quickstart_prediction_start__
+            :end-before: __pt_quickstart_prediction_end__
+
+    .. group-tab:: TensorFlow
+
+        .. literalinclude:: ./doc_code/tf_quick_start.py
+            :language: python
+            :start-after: __tf_quickstart_prediction_start__
+            :end-before: __tf_quickstart_prediction_end__
+
+.. note::
```

Comment:
Suggest either using functions or actors, but not mentioning both, because this is a quickstart, not a comprehensive guide.

Reply:
very good point, moved it to the advanced section